lib/views/location_card.dart

```diff
@@ -12,117 +18,143 @@ class LocationCard extends StatefulWidget {
 }
 
 class _LocationCardState extends State<LocationCard> {
-  Future<bool> _fetchLocationData() {
-    return widget.location.fetchData();
+  late Future<WeatherData> _weatherData;
+
+  @override
+  void initState() {
+    _weatherData = widget.location.fetchData();
+    super.initState();
+  }
+
+  String _airQuality(double no2Concentration) {
+    if (no2Concentration > 400) {
+      return 'Very poor';
+    } else if (no2Concentration > 200) {
+      return 'Poor';
+    } else if (no2Concentration > 100) {
+      return 'Moderate';
+    } else if (no2Concentration > 50) {
+      return 'Fair';
+    } else {
+      return 'Good';
+    }
   }
 
   @override
   Widget build(BuildContext context) {
-    return FutureBuilder<bool>(
-      future: _fetchLocationData(),
+    List<String> tabs = ['Forecast', 'Air quality'];
+    return FutureBuilder<WeatherData>(
+      future: _weatherData,
       builder: (context, snapshot) {
         if (snapshot.hasData) {
-          return SafeArea(
-            child: Container(
-              decoration: BoxDecoration(
-                image: DecorationImage(
-                  image: AssetImage(
-                    (widget.location.isDayTime ?? true)
-                        ? 'assets/day.jpg'
-                        : 'assets/night.jpg',
-                  ),
-                  fit: BoxFit.cover,
+          return Container(
+            decoration: BoxDecoration(
+              image: DecorationImage(
+                image: AssetImage(
+                  (snapshot.data!.isDayTime)
+                      ? 'assets/day.jpg'
+                      : 'assets/night.jpg',
```

Comment:
Nit:
```suggestion
                    ? 'assets/day.jpg'
                    : 'assets/night.jpg',
```

Reply:
For some reason when it gets formatted it uses double indentation, not sure why